crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -94,9 +94,17 @@ impl CoinReadApiServer for CoinReadApi {
             let coin_type_tag = parse_to_type_tag(coin_type)?;
 
             let cursor = match cursor {
-                Some(c) => (coin_type_tag.to_string(), c),
+                Some(c) => {
+                    let obj = self.internal.get_object(&c).await?.ok_or_else(|| {
+                        SuiRpcInputError::GenericInvalid("cursor not found".to_string())
+                    })?;
+                    let coin = obj.as_coin_maybe().ok_or_else(|| {
+                        SuiRpcInputError::GenericInvalid("cursor is not a coin".to_string())
+                    })?;
+                    (coin_type_tag.to_string(), !coin.balance.value(), c)
+                }
```

Comment:
Yeah :( 

I had two options:
1. break the cursor to now take the balance
2. do this

This seemed the least bad option.

It does highlight that cursors like this should be less defined so that they can change if need be in the future